libs/core/langchain_core/output_parsers/pydantic.py

```diff
@@ -1,34 +1,56 @@
 import json
-from typing import Generic, List, Type, TypeVar
+from typing import Generic, List, Type, TypeVar, Union
 
 from langchain_core.exceptions import OutputParserException
 from langchain_core.output_parsers import JsonOutputParser
 from langchain_core.outputs import Generation
 from langchain_core.pydantic_v1 import BaseModel, ValidationError
+from langchain_core.pydantic_v2 import BaseModel as V2BaseModel
+from langchain_core.pydantic_v2 import ValidationError as V2ValidationError
+from langchain_core.utils.pydantic import PYDANTIC_MAJOR_VERSION
 
-TBaseModel = TypeVar("TBaseModel", bound=BaseModel)
+PydanticBaseModel = BaseModel
+if PYDANTIC_MAJOR_VERSION == 2:
+    PydanticBaseModel = Union[BaseModel, V2BaseModel]  # type: ignore
+
+TBaseModel = TypeVar("TBaseModel", bound=PydanticBaseModel)
 
 
 class PydanticOutputParser(JsonOutputParser, Generic[TBaseModel]):
     """Parse an output using a pydantic model."""
 
     pydantic_object: Type[TBaseModel]
-    """The pydantic model to parse.
-    
-    Attention: To avoid potential compatibility issues, it's recommended to use
-        pydantic <2 or leverage the v1 namespace in pydantic >= 2.
-    """
+    """The pydantic model to parse."""
+
+    def _parse_obj(self, obj: dict) -> TBaseModel:
+        if PYDANTIC_MAJOR_VERSION == 2 and issubclass(
+            self.pydantic_object, V2BaseModel
+        ):
+            try:
```

Comment:
The validation error can be imported here locally -- the import is cached so there's no issue in terms of speed

```python
if VERSION = 2:
   if issubclass(self.pydantic_object, V1):
      .....
       except pydantic.v1.ValidationError:
   else:
       except pydantic.ValidationError:
```